packages/tg3/README.md

```diff
@@ -0,0 +1,131 @@
+
+- K of N immutable multisig. One key proposes a set of messages,
+  after K-1 others approve it, it can be executed with the
+  multisig address.
+- K of N flexible, mutable multisig. Like above, but with
+  multiple contracts. One contract stores the group, which is
+  referenced from multiple multisig contracts (which in turn
+  implement tg3). One tg3 contracts is able to update the
+  group content (maybe needing 67% vote). Other tg3 contracts
+  may hold tokens, staking rights, etc with various execution
+  thresholds, all controlled by one group. (Group interface
+  and updating them are defined in tg4 spec).
+
+This should fit in this interface (possibly with some
+extensions for pieces, but the usage should look the
+same externally):
+
+- Points weighted voting. People have some arbitrary voting
+  power represented by points. There is a vote threshold
+  to execute messages. The voting set is dynamic. This has a similar
+  "propose, approve, execute" flow, but we will need to  support clear YES/NO
+  votes and quora not just absolute thresholds.
+
+The common denominator is that they allow you to propose
+arbitrary `CosmosMsg` to a contract, and allow a series
+of votes/approvals to determine if it can be executed,
+as well as a final step to execute any approved proposal once.
+
+## Base
+
+The following interfaces must be implemented for all tg3
+contracts. Note that updating the members of the voting
+contract is not contained here (one approach is defined in cw4).
+Also, how to change the threshold rules (if at all) is not
+standardized. Those are considered admin tasks, and the common
+API is designed for standard usage, as that is where we can
+standardize the most tooling without limiting more complex
+governance controls.
+
+### Messages
+
+`Propose{title, description, msgs, earliest, latest}` - This accepts
+`Vec<CosmosMsg>` and creates a new proposal. This will return
+an auto-generated ID in the `Data` field (and the logs) that
+can be used to reference the proposal later.
+
+If the Proposer is a valid voter on the proposal, this will imply a Yes vote by
+the Proposer for a faster workflow, especially useful in eg. 2 of 3
+or 3 of 5 multisig, we don't need to propose in one block, get result,
+and vote in another block.
+
+Earliest and latest are optional and can request the first
+and last height/time that we can try `Execute`. For a vote,
+we may require at least 2 days to pass, but no more than 7.
+This is optional and even if set, may be modified by the contract
+(overriding or just enforcing min/max/default values).
+
+Many implementations will want to restrict who can propose.
+Maybe only people in the voting set. Maybe there is some
+deposit to be made along with the proposal. This is not
+in the spec but left open to the implementation.
+
+`Vote{proposal_id, vote}` - Given a proposal_id, you can
+vote yes, no, abstain or veto. Each signed may have a
+different "weight" in the voting and they apply their
```

Comment:
-> points

Reply:
Ehh, grepped for READMEs and still miss it :/